datafusion/physical-expr/src/aggregate/first_last.rs

```diff
@@ -237,13 +231,17 @@ impl Accumulator for FirstValueAccumulator {
         };
         if !ordered_states[0].is_empty() {
             let first_row = get_row_at_idx(&ordered_states, 0)?;
-            let first_ordering = &first_row[1..];
+            // When collecting orderings, we exclude the is_set flag from the state.
+            let first_ordering = &first_row[1..is_set_idx];
             let sort_options = get_sort_options(&self.ordering_req);
             // Either there is no existing value, or there is an earlier version in new data.
             if !self.is_set
                 || compare_rows(first_ordering, &self.orderings, &sort_options)?.is_lt()
```

Comment:
I am wondering if we need to do anything about `compare_rows(first_ordering, &self.orderings, &sort_options)` since now `first_ordering` is shorter?

Reply:
It is not shorter. Actually, this fix makes them same length. Previously `first_ordering` has more one element (`is_set`).

Reply:
Actually I found the only time `self.orderings` can be longer is right after the creation by `try_new()`. But at the time it is `self.is_set = false`, so it will not hit `compare_rows`. Then `self.orderings` is updated by `first_ordering`